java/src/jmri/jmrix/can/cbus/node/CbusNodeEventManager.java

```diff
@@ -502,9 +499,20 @@ protected void teachNewEvLoop(){
             teachEventsComplete();
             return;
         }
-        
+
         CbusNodeEvent wholeEvent = eventsToTeachArray.get(nextEvInArray);
-        
+        log.debug("event variable array length: {}", wholeEvent.getEvVarArray().length);
+        log.debug("Number of event vars from node parameters: {}",_node.getNodeParamManager().getParameter(CbusNodeConstants.EV_PER_EN_IDX) );
+        if (wholeEvent.getEvVarArray().length < nextEvVar ){
```

Comment:
I don't know if it will ever happen, but if `wholeEvent.getEvVarArray()` is empty, this code will loop forever.

The first constructor of `CbusNodeEvent` has the code:
`_evVarArr = new int[Math.max(maxEvVar, 0)];`

So if `maxEvVar` is zero or negative, the array will be empty.

Reply:
After looking at the code again, I think I was wrong and that there is no risk of endless loop.

Reply:
I'll be adding some unit tests surrounding this, no harm in checking.